tgui/packages/tgui/interfaces/PowerMonitor/Smes.tsx

```diff
@@ -0,0 +1,103 @@
+import { useBackend, useSharedState } from '../../backend';
+import { Box, Chart, LabeledList, Stack, Table } from '../../components';
+import { formatPower } from '../../format';
+import { PowerMonitorSmesData, PowerMonitorSmesItemData } from './type';
+
+export const PowerMonitorSmesGlobal = (_props, context) => {
+  const { data } = useBackend<PowerMonitorSmesData>(context);
+
+  const availableHistory = data.history.map((v) => v[0]);
+  const availableHistoryData = availableHistory.map((v, i) => [i, v]);
+
+  const loadHistory = data.history.map((v) => v[1]);
+  const loadHistoryData = loadHistory.map((v, i) => [i, v]);
+
+  const max = Math.max(...availableHistory, ...loadHistory);
+
+  return (
+    <Stack fill>
+      <Stack.Item width="50%">
+        <LabeledList>
+          <LabeledList.Item label="Engine Output">{formatPower(data.available)}</LabeledList.Item>
+        </LabeledList>
+        <Chart.Line
+          mt="5px"
+          height="5em"
+          data={availableHistoryData}
+          rangeX={[0, availableHistoryData.length - 1]}
+          rangeY={[0, max]}
+          strokeColor="rgba(1, 184, 170, 1)"
+          fillColor="rgba(1, 184, 170, 0.25)"
+        />
+      </Stack.Item>
+      <Stack.Item width="50%">
+        <LabeledList>
+          <LabeledList.Item label="SMES/PTL Draw">{formatPower(data.load)}</LabeledList.Item>
+        </LabeledList>
+        <Chart.Line
+          mt="5px"
+          height="5em"
+          data={loadHistoryData}
+          rangeX={[0, loadHistoryData.length - 1]}
+          rangeY={[0, max]}
+          strokeColor="rgba(1, 184, 170, 1)"
+          fillColor="rgba(1, 184, 170, 0.25)"
+        />
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+export const PowerMonitorSmesTableHeader = (props, context) => {
+  return (
+    <>
+      <Table.Cell header>Area</Table.Cell>
+      <Table.Cell header>Stored Power</Table.Cell>
+      <Table.Cell header>Charging</Table.Cell>
+      <Table.Cell header>Input</Table.Cell>
+      <Table.Cell header>Output</Table.Cell>
+      <Table.Cell header>Active</Table.Cell>
+      <Table.Cell header>Load</Table.Cell>
+    </>
+  );
+};
+
+export const PowerMonitorSmesTableRows = (props, context) => {
+  const { data } = useBackend<PowerMonitorSmesData>(context);
+
+  return (
+    <>
+      {data.units.map((unit) => (
+        <PowerMonitorSmesTableRow key={unit[0]} unit={unit} />
+      ))}
+    </>
+  );
+};
+
+type PowerMonitorSmesTableRowProps = {
+  unit: PowerMonitorSmesItemData;
+};
+
+const PowerMonitorSmesTableRow = (props: PowerMonitorSmesTableRowProps, context) => {
+  const { unit } = props;
+  const [ref, stored, charging, input, output, online, load] = unit;
```

Comment:
Not a huge fan of lists of things whose meaning is derived via index, this feels like it should be an object with key/value pairs to me. I appreciate that this would increase the data size quite significantly, however, so would probably prefer this as you have it, with a comment above to explain the reasoning.